src/java/org/apache/cassandra/net/SocketFactory.java

```diff
@@ -237,6 +239,29 @@ static String encryptionLogStatement(EncryptionOptions options)
         return "enabled (" + encryptionType + ')';
     }
 
+    static String encryptionLogStatement(Channel channel, EncryptionOptions options)
+    {
+        if (options == null)
+            return "disabled";
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("enabled (factory=");
+        sb.append(SSLFactory.openSslIsAvailable() ? "openssl" : "jdk");
+
+        SslHandler sslHandler = channel.pipeline().get(SslHandler.class);
```

Comment:
you pass null to channel but don't have any checks against it; in testing this patch found a case where this causes a NPE

```
ERROR [main] 2020-07-30 16:00:56,584 CassandraDaemon.java:800 - Exception encountered during startup
java.lang.NullPointerException: null
  at org.apache.cassandra.net.SocketFactory.encryptionLogStatement(SocketFactory.java:251)
  at org.apache.cassandra.net.InboundConnectionSettings.toString(InboundConnectionSettings.java:87)
  at java.base/java.lang.String.valueOf(String.java:2951)
  at java.base/java.lang.StringBuilder.append(StringBuilder.java:168)
  at org.apache.cassandra.net.InboundConnectionInitiator.bind(InboundConnectionInitiator.java:129)
```

Reply:
fixed